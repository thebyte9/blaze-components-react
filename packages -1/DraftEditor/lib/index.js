"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
exports.__esModule = true;
var utils_1 = __importDefault(require("@blaze-react/utils"));
var draft_js_1 = require("draft-js");
var react_1 = __importStar(require("react"));
var BlockControls_1 = __importDefault(require("./BlockControls"));
var InlineControls_1 = __importDefault(require("./InlineControls"));
var DraftEditor = function (_a) {
    var _b = _a.utils, classNames = _b.classNames, ErrorMessage = _b.ErrorMessage, onChange = _a.onChange, name = _a.name, value = _a.value, error = _a.error, validationMessage = _a.validationMessage, attrs = __rest(_a, ["utils", "onChange", "name", "value", "error", "validationMessage"]);
    var draftHandledValue = "handled";
    var draftNotHandledValue = "not-handled";
    var _c = react_1.useState(draft_js_1.EditorState.createEmpty()), editorState = _c[0], setEditorState = _c[1];
    react_1.useEffect(function () {
        if (value) {
            var rawObjectValue = JSON.parse(value);
            var state = draft_js_1.EditorState.createWithContent(draft_js_1.convertFromRaw(rawObjectValue));
            setEditorState(state);
        }
    }, []);
    var onEditorChange = function (newEditorState) {
        setEditorState(newEditorState);
        var currentContent = newEditorState.getCurrentContent();
        var rawValue = draft_js_1.convertToRaw(currentContent);
        var rawValueString = JSON.stringify(rawValue);
        var eventFormat = {
            event: {
                target: {
                    name: name,
                    value: rawValueString
                }
            }
        };
        if (onChange) {
            onChange(eventFormat);
        }
    };
    var toggleBlockType = function (blockType) {
        return onEditorChange(draft_js_1.RichUtils.toggleBlockType(editorState, blockType));
    };
    var toggleInlineStyle = function (inlineStyle) {
        return onEditorChange(draft_js_1.RichUtils.toggleInlineStyle(editorState, inlineStyle));
    };
    var handleKeyCommand = function (command) {
        var newState = draft_js_1.RichUtils.handleKeyCommand(editorState, command);
        if (newState) {
            onEditorChange(newState);
            return draftHandledValue;
        }
        return draftNotHandledValue;
    };
    var contentState = editorState.getCurrentContent();
    var isUnstyled = contentState
        .getBlockMap()
        .first()
        .getType() !== "unstyled";
    var hasTextAndUnstyled = !contentState.hasText() && isUnstyled;
    var editorClassName = classNames("custom-DraftEditor-editor", {
        "custom-DraftEditor-hidePlaceholder": hasTextAndUnstyled
    });
    var getBlockStyle = function (block) {
        var isBlockquote = block.getType() === "blockquote";
        return classNames({
            "custom-DraftEditor-blockquote": isBlockquote
        });
    };
    return (react_1["default"].createElement("div", { className: "custom-DraftEditor-root" },
        react_1["default"].createElement(BlockControls_1["default"], { editorState: editorState, onToggle: toggleBlockType }),
        react_1["default"].createElement(InlineControls_1["default"], { editorState: editorState, onToggle: toggleInlineStyle }),
        react_1["default"].createElement("div", { className: editorClassName },
            react_1["default"].createElement(draft_js_1.Editor, __assign({ blockStyleFn: getBlockStyle, editorState: editorState, onChange: onEditorChange, handleKeyCommand: handleKeyCommand }, attrs))),
        error && react_1["default"].createElement(ErrorMessage, { message: validationMessage })));
};
DraftEditor.defaultProps = {
    error: false,
    name: "editor",
    validationMessage: "This field is required"
};
exports["default"] = utils_1["default"](DraftEditor);
