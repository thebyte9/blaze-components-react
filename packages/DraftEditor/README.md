## Description

Draft.js is a framework for building rich text editors in React, makes it easy to build any type of rich text input, whether you're just looking to support a few inline text styles or building a complex text editor for composing long-form articles.

In Draft.js, everything is customizable check de documentation in the following link:

[Learn more about Draft](https://draftjs.org/docs/getting-started.html)
