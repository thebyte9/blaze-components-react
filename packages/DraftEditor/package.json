{
  "name": "@blaze-react/drafteditor",
  "version": "0.8.0-alpha.13",
  "description": "Draft.js is a framework for building rich text editors in React",
  "main": "./dist/main.js",
  "scripts": {
    "build": "NODE_ENV=production webpack --config webpack.config.js",
    "build-storybook": "build-storybook -c .storybook -o docs-build",
    "lint": "eslint '*/**/*.{ts,tsx}' --ignore-path ../../.eslintignore --fix",
    "test": "jest --passWithNoTests",
    "test:coverage": "jest --coverage --passWithNoTests",
    "pre-publish": "npm run build && npm run lint && npm run test"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "peerDependencies": {
    "draft-js-export-html": "^1.4.1",
    "html-react-parser": "^1.2.6",
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  },
  "devDependencies": {
    "@babel/cli": "^7.4.4",
    "@babel/core": "^7.4.5",
    "@types/draft-js": "^0.11.1",
    "compression-webpack-plugin": "^7.1.2",
    "postcss-preset-env": "^6.7.0",
    "webpack": "^5.28.0",
    "webpack-cli": "^4.6.0"
  },
  "gitHead": "91fa52db7188d6090f03041963d00fa3405a7dcb"
}
