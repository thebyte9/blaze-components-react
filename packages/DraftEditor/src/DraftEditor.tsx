import { Editor, EditorState } from 'draft-js';

import { EMPTY_STRING } from './constants';
import { ISelectionRect } from './helpers/inline-toolbar';
import InlineToolbar from './InlineToolbar';
import PropTypes from 'prop-types';
import React from 'react';
import { getSelectedText, Rect } from './helpers';
interface IEditor {
  editorState: EditorState;
  dispatch: any;
  selectionRect: ISelectionRect;
  showAddLinkModal: (value: boolean) => void;
  handleChange: any;
  editor: any;
  handleKeyCommand: any;
  inlineToolbar: any;
  showInlineToolbar: any;
  save: any;
  styleMap: any;
  myKeyBindingFn: any;
  customBlockStyle: any;
  customBlockRenderer: any;
  handleOnBlur: any;
  handleOnFocus: any;
}

const DraftEditor = ({
  editorState,
  dispatch,
  selectionRect,
  showAddLinkModal,
  handleChange,
  editor,
  handleKeyCommand,
  inlineToolbar,
  showInlineToolbar,
  save,
  styleMap,
  myKeyBindingFn,
  customBlockStyle,
  customBlockRenderer,
  handleOnBlur,
  handleOnFocus,
}: IEditor) => {
  return (
    <div
      className="editor-view__textblock--editor"
      ref={(el) => {
        if (!el) return;
        Rect.rect = el.getBoundingClientRect();
      }}
    >
      <InlineToolbar
        editorState={editorState}
        dispatch={dispatch}
        selectionRect={selectionRect}
        showAddLinkModal={showAddLinkModal}
        onChange={(state) => handleChange(state)}
        visible={inlineToolbar && getSelectedText(editorState) !== EMPTY_STRING}
      />
      <Editor
        ref={editor}
        editorState={editorState}
        onChange={(state) => handleChange(state)}
        handleKeyCommand={(command) =>
          handleKeyCommand(command, editorState, save, dispatch, showAddLinkModal, showInlineToolbar)
        }
        customStyleMap={styleMap}
        keyBindingFn={myKeyBindingFn}
        blockStyleFn={customBlockStyle}
        blockRendererFn={customBlockRenderer}
        onBlur={(e) => handleOnBlur(e)}
        onFocus={(e) => handleOnFocus(e)}
      />
    </div>
  );
};

DraftEditor.propTypes = {
  editorState: PropTypes.object.isRequired,
  dispatch: PropTypes.func.isRequired,
  selectionRect: PropTypes.object.isRequired,
  showAddLinkModal: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  editor: PropTypes.object.isRequired,
  handleKeyCommand: PropTypes.func.isRequired,
  inlineToolbar: PropTypes.bool.isRequired,
  showInlineToolbar: PropTypes.func.isRequired,
  save: PropTypes.func.isRequired,
  styleMap: PropTypes.object.isRequired,
  myKeyBindingFn: PropTypes.func.isRequired,
  customBlockStyle: PropTypes.func.isRequired,
  customBlockRenderer: PropTypes.func.isRequired,
  handleOnBlur: PropTypes.func.isRequired,
  handleOnFocus: PropTypes.func.isRequired,
};

export default DraftEditor;
