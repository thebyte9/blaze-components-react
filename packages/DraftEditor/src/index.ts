import DraftEditorWrapper from './DraftEditorWrapper';
import parseTextblock from './parser';
export { DraftEditorWrapper, parseTextblock };
