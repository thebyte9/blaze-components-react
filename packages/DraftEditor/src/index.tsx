import withUtils from "@blaze-react/utils";
import eventBus from "./eventBus";

import {
  AtomicBlockUtils,
  ContentBlock,
  ContentState,
  convertFromRaw,
  convertToRaw,
  DraftEditorCommand,
  DraftHandleValue,
  EditorBlock,
  EditorState,
  RichUtils
} from "draft-js";
import Editor from "draft-js-plugins-editor";
import React, { FunctionComponent, useEffect, useRef, useState } from "react";
import {
  ATOMIC,
  BLOCKQUOTE,
  HANDLED,
  NOT_HANDLED,
  UNSTYLED
} from "./constants";
import { DraftPlugins, plugins } from "./DraftPlugins";
import { CustomDraftPlugins } from "./DraftPlugins/CustomPlugins";
import decorator from "./DraftPlugins/CustomPlugins/decorator";
import { AddImageAttributes } from "./DraftPlugins/CustomPlugins/ImageControl";
import { IDraftEditorProps } from "./interfaces";
import { addButtonToAlignmentToolContainer, getEditorHeight } from "./utils";

const blockRenderer = (contentBlock: any) => {
  const type = contentBlock.getType();

  if (type === ATOMIC) {
    return {
      component: Component,
      editable: true,
      props: {}
    };
  }
  return "";
};

const Component = (props: any) => (
  <pre>
    <code>
      <EditorBlock {...props} />
    </code>
  </pre>
);

const DraftEditor: FunctionComponent<IDraftEditorProps> = ({
  utils: { classNames, ErrorMessage },
  onChange,
  name,
  value,
  error,
  validationMessage,
  unSelectedText,
  selectedImages,
  handleLibraryClick,
  ...attrs
}): JSX.Element => {
  const draftHandledValue: DraftHandleValue = HANDLED;
  const draftNotHandledValue: DraftHandleValue = NOT_HANDLED;

  const [editorState, setEditorState] = useState<EditorState>(
    EditorState.createWithContent(
      EditorState.createEmpty().getCurrentContent(),
      decorator
    )
  );
  const [editorHeight, setEditorHeight] = useState<any>({});
  const [imageAttributesStatus, setImageAttributesStatus] = useState<boolean>(
    false
  );
  const [imageAttributesData, setImageAttributesData] = useState<any>({
    focusedImageURL: null,
    images: []
  });
  const inputEl = useRef<any>(null);
  const globalRef = useRef<any>(null);

  useEffect((): void => {
    let initialEditorState = EditorState.createEmpty().getCurrentContent();
    let images: any = [];

    if (value) {
      const parsedValue = JSON.parse(value);
      images =
        parsedValue.imageAttributes instanceof Array
          ? parsedValue.imageAttributes
          : [];
      setImageAttributesData({
        focusedImageURL: null,
        images
      });
      initialEditorState = convertFromRaw(parsedValue);
    }

    const state: EditorState = EditorState.createWithContent(
      initialEditorState,
      decorator
    );
    setEditorState(state);
    onEditorChange(state);
    calculateEditorHeight(500);
    addButtonToAlignmentToolContainer(globalRef.current);
    handleEditImageEvent(images);
  }, []);

  const closeImageAttributesModal = () => setImageAttributesStatus(false);

  const saveImageAttributes = (imageAttributes: any) => {
    setImageAttributesData(imageAttributes);
    onEditorChange(editorState, imageAttributes.images);
    handleEditImageEvent(imageAttributes.images);
  };

  useEffect((): void => {
    calculateEditorHeight();
  }, [editorState]);

  const handleEditImageEvent = (images: any) => {
    eventBus.$on("editImageAttributes", focusedImageURL => {
      setImageAttributesStatus(true);
      setImageAttributesData({
        focusedImageURL,
        images
      });
    });
  };

  const calculateEditorHeight = (time = 0) =>
    setTimeout(() => setEditorHeight(getEditorHeight(inputEl.current)), time);

  const onEditorChange = (
    newEditorState: EditorState,
    imagesAttr?: any[]
  ): void => {
    const currentContent = newEditorState.getCurrentContent();
    const rawValue = convertToRaw(currentContent);

    const blocks = rawValue.blocks.map(block => {
      if (block.type === "atomic" && !!block.text.trim()) {
        block.text = block.text.replace(/\s+/g, " ");
      }
      return block;
    });

    rawValue.blocks = blocks;

    const rawValueString = JSON.stringify({
      ...rawValue,
      imageAttributes: imagesAttr instanceof Array ? imagesAttr : []
    });

    const eventFormat = {
      event: {
        target: {
          name,
          value: rawValueString
        }
      }
    };
    if (onChange) {
      setEditorState(newEditorState);
      onChange(eventFormat);
    }
  };

  const contentState: ContentState = editorState.getCurrentContent();

  const isUnstyled: boolean =
    contentState
      .getBlockMap()
      .first()
      .getType() !== UNSTYLED;

  const hasTextAndUnstyled: boolean = !contentState.hasText() && isUnstyled;

  const editorClassName: string = classNames("custom-DraftEditor-editor", {
    "custom-DraftEditor-hidePlaceholder": hasTextAndUnstyled
  });

  const getBlockStyle = (block: ContentBlock): string => {
    const isBlockquote: boolean = block.getType() === BLOCKQUOTE;
    return classNames({
      "custom-DraftEditor-blockquote": isBlockquote
    });
  };

  const handleKeyCommand = (command: DraftEditorCommand): DraftHandleValue => {
    const newState: EditorState = RichUtils.handleKeyCommand(
      editorState,
      command
    );
    if (newState) {
      onEditorChange(newState);
      return draftHandledValue;
    }
    return draftNotHandledValue;
  };

  const insertBlock = () => {
    const ccontentState = editorState.getCurrentContent();

    const contentStateWithEntity = ccontentState.createEntity(
      "CODE",
      "MUTABLE"
    );

    const entityKey = contentStateWithEntity.getLastCreatedEntityKey();
    const newEditorState = EditorState.set(editorState, {
      currentContent: contentStateWithEntity
    });

    setEditorState(
      AtomicBlockUtils.insertAtomicBlock(newEditorState, entityKey, " ")
    );
  };

  return (
    <div className="custom-DraftEditor-root" ref={globalRef}>
      <CustomDraftPlugins
        editorState={editorState}
        selectedImages={selectedImages}
        handleLibraryClick={handleLibraryClick}
        unSelectedText={unSelectedText}
        onEditorChange={onEditorChange}
        toggleDraftEditor={insertBlock}
      />
      {imageAttributesStatus && (
        <AddImageAttributes
          imageAttributesData={imageAttributesData}
          saveImageAttributes={saveImageAttributes}
          closeImageAttributesModal={closeImageAttributesModal}
        />
      )}

      <div className={editorClassName} style={editorHeight}>
        <Editor
          ref={inputEl}
          handleKeyCommand={handleKeyCommand}
          blockStyleFn={getBlockStyle}
          editorState={editorState}
          onChange={onEditorChange}
          blockRendererFn={blockRenderer}
          plugins={plugins}
          {...attrs}
        />
        <DraftPlugins />
      </div>
      {error && <ErrorMessage message={validationMessage} />}
    </div>
  );
};

DraftEditor.defaultProps = {
  error: false,
  name: "editor",
  selectedImages: [],
  unSelectedText: "Make sure you have a text selected",
  validationMessage: "This field is required"
};

export default withUtils(DraftEditor);
