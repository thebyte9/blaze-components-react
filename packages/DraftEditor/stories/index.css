.story-wrapper {
  margin-top: 50px;
}
