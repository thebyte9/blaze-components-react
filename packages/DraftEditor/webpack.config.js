const path = require('path');
const ForkTsCheckerWebpackPlugin = require('fork-ts-checker-webpack-plugin');
const ESLintPlugin = require('eslint-webpack-plugin');
const MiniCssExtractPlugin = require('mini-css-extract-plugin');
const CssMinimizerPlugin = require('css-minimizer-webpack-plugin');

module.exports = {
  mode: 'production',
  devtool: 'cheap-module-source-map',
  entry: {
    index: {
      import: './src/index.ts',
      dependOn: 'shared',
    },
    parser: {
      import: './src/parser/index.tsx',
      dependOn: 'shared',
    },
    shared: 'draft-js',
  },
  output: {
    path: path.resolve(__dirname, 'dist'),
    filename: '[name].js',
    clean: true,
  },
  optimization: {
    minimize: true,
    minimizer: [`...`, new CssMinimizerPlugin()],
    usedExports: true,
    removeAvailableModules: true,
    concatenateModules: true,
    runtimeChunk: 'single',
  },
  resolve: {
    extensions: ['.js', '.json', '.ts', '.tsx'],
  },
  module: {
    rules: [
      {
        test: /\.(ts|js)x?$/i,
        exclude: /node_modules/,
        use: {
          loader: 'babel-loader',
          options: {
            presets: [
              '@babel/preset-env',
              '@babel/preset-typescript',
              [
                '@babel/preset-react',
                {
                  runtime: 'automatic',
                },
              ],
            ],
          },
        },
      },
      {
        test: /\.scss$/,
        use: [
          MiniCssExtractPlugin.loader,
          'css-loader',
          {
            loader: 'postcss-loader',
            options: {
              postcssOptions: {
                plugins: [['postcss-preset-env']],
              },
            },
          },
          'sass-loader',
        ],
      },
    ],
  },
  plugins: [
    new ForkTsCheckerWebpackPlugin({
      async: false,
    }),
    new ESLintPlugin({
      extensions: ['js', 'jsx', 'ts', 'tsx'],
    }),
    new MiniCssExtractPlugin({
      filename: '[name].[contenthash].css',
    }),
  ],
};
