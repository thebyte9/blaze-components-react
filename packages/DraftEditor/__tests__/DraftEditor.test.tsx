import DraftEditor from '../src/DraftEditor';
import React from 'react';
import { render } from '@testing-library/react';

declare let window: any;

jest.mock('../src/DraftEditor', () => () => <div>Draft Editor</div>);

describe('DraftEditor component', () => {
  window.getSelection = () => {
    return {
      removeAllRanges: () => ({}),
    };
  };

  beforeEach(() => {
    Object.defineProperty(window, 'localStorage', {
      value: {
        getItem: jest.fn(() => '{ "name":"John", "age":30, "city":"New York"}'),
        setItem: jest.fn(() => '{ "name":"John", "age":30, "city":"New York"}'),
      },
      writable: true,
    });
  });

  test('should be defined and renders correctly (snapshot)', () => {
    const props = {
      handleChange: jest.fn(),
      handleOnFocus: jest.fn(),
      handleOnBlur: jest.fn(),
      handleOnFocus: jest.fn(),
      handleOnBlur: jest.fn(),
      handleKeyCommand: jest.fn(),
      customBlockRenderer: jest.fn(),
      showInlineToolbar: jest.fn(),
      inlineToolbar: false,
      save: jest.fn(),
      styleMap: {},
      myKeyBindingFn: jest.fn(),
      customBlockStyle: jest.fn(),
      customBlockStyle: jest.fn(),
      editor: React.createRef(),
      showAddLinkModal: jest.fn(),
      selectionRect: {
        top: 0,
        bottom: 0,
        right: 0,
        left: 0,
      },
      editorState: {
        getCurrentContent: jest.fn(),
        getSelection: jest.fn(),
        mustForceSelection: jest.fn(),
        getDecorator: jest.fn(),
        getDirectionMap: jest.fn(),
      },
      dispatch: jest.fn(),
    };

    const { asFragment, rerender } = render(<DraftEditor {...props} />);

    const testValueJSON =
      '{"blocks":[{"key":"ai4n8","text":"testing","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}}],"entityMap":{}}';

    rerender(<DraftEditor value={testValueJSON} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
