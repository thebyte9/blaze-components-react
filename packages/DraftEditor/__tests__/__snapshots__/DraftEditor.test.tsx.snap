// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DraftEditor component should be defined and renders correctly (snapshot) 1`] = `
<Fragment>
  <BlockControls
    editorState={
      EditorState {
        "_immutable": Immutable.Record {
          "allowUndo": true,
          "currentContent": Immutable.Record {
            "entityMap": Object {
              "__add": [Function],
              "__create": [Function],
              "__get": [Function],
              "__getLastCreatedEntityKey": [Function],
              "__mergeData": [Function],
              "__replaceData": [Function],
              "add": [Function],
              "create": [Function],
              "get": [Function],
              "getLastCreatedEntityKey": [Function],
              "mergeData": [Function],
              "replaceData": [Function],
            },
            "blockMap": Immutable.OrderedMap {
              "3cmdi": Immutable.Record {
                "key": "3cmdi",
                "type": "unstyled",
                "text": "",
                "characterList": Immutable.List [],
                "depth": 0,
                "data": Immutable.Map {},
              },
            },
            "selectionBefore": Immutable.Record {
              "anchorKey": "3cmdi",
              "anchorOffset": 0,
              "focusKey": "3cmdi",
              "focusOffset": 0,
              "isBackward": false,
              "hasFocus": false,
            },
            "selectionAfter": Immutable.Record {
              "anchorKey": "3cmdi",
              "anchorOffset": 0,
              "focusKey": "3cmdi",
              "focusOffset": 0,
              "isBackward": false,
              "hasFocus": false,
            },
          },
          "decorator": null,
          "directionMap": Immutable.OrderedMap {
            "3cmdi": "LTR",
          },
          "forceSelection": false,
          "inCompositionMode": false,
          "inlineStyleOverride": null,
          "lastChangeType": null,
          "nativelyRenderedContent": null,
          "redoStack": Immutable.Stack [],
          "selection": Immutable.Record {
            "anchorKey": "3cmdi",
            "anchorOffset": 0,
            "focusKey": "3cmdi",
            "focusOffset": 0,
            "isBackward": false,
            "hasFocus": false,
          },
          "treeMap": Immutable.OrderedMap {
            "3cmdi": Immutable.List [
              Immutable.Record {
                "start": 0,
                "end": 0,
                "decoratorKey": null,
                "leaves": Immutable.List [
                  Immutable.Record {
                    "start": 0,
                    "end": 0,
                  },
                ],
              },
            ],
          },
          "undoStack": Immutable.Stack [],
        },
      }
    }
    onToggle={[Function]}
  />
  <InlineControls
    editorState={
      EditorState {
        "_immutable": Immutable.Record {
          "allowUndo": true,
          "currentContent": Immutable.Record {
            "entityMap": Object {
              "__add": [Function],
              "__create": [Function],
              "__get": [Function],
              "__getLastCreatedEntityKey": [Function],
              "__mergeData": [Function],
              "__replaceData": [Function],
              "add": [Function],
              "create": [Function],
              "get": [Function],
              "getLastCreatedEntityKey": [Function],
              "mergeData": [Function],
              "replaceData": [Function],
            },
            "blockMap": Immutable.OrderedMap {
              "3cmdi": Immutable.Record {
                "key": "3cmdi",
                "type": "unstyled",
                "text": "",
                "characterList": Immutable.List [],
                "depth": 0,
                "data": Immutable.Map {},
              },
            },
            "selectionBefore": Immutable.Record {
              "anchorKey": "3cmdi",
              "anchorOffset": 0,
              "focusKey": "3cmdi",
              "focusOffset": 0,
              "isBackward": false,
              "hasFocus": false,
            },
            "selectionAfter": Immutable.Record {
              "anchorKey": "3cmdi",
              "anchorOffset": 0,
              "focusKey": "3cmdi",
              "focusOffset": 0,
              "isBackward": false,
              "hasFocus": false,
            },
          },
          "decorator": null,
          "directionMap": Immutable.OrderedMap {
            "3cmdi": "LTR",
          },
          "forceSelection": false,
          "inCompositionMode": false,
          "inlineStyleOverride": null,
          "lastChangeType": null,
          "nativelyRenderedContent": null,
          "redoStack": Immutable.Stack [],
          "selection": Immutable.Record {
            "anchorKey": "3cmdi",
            "anchorOffset": 0,
            "focusKey": "3cmdi",
            "focusOffset": 0,
            "isBackward": false,
            "hasFocus": false,
          },
          "treeMap": Immutable.OrderedMap {
            "3cmdi": Immutable.List [
              Immutable.Record {
                "start": 0,
                "end": 0,
                "decoratorKey": null,
                "leaves": Immutable.List [
                  Immutable.Record {
                    "start": 0,
                    "end": 0,
                  },
                ],
              },
            ],
          },
          "undoStack": Immutable.Stack [],
        },
      }
    }
    onToggle={[Function]}
  />
  <DraftEditor
    blockRenderMap={
      Immutable.Map {
        "ordered-list-item": Object {
          "element": "li",
          "wrapper": <ol
            className="public-DraftStyleDefault-ol"
          />,
        },
        "header-six": Object {
          "element": "h6",
        },
        "header-four": Object {
          "element": "h4",
        },
        "header-one": Object {
          "element": "h1",
        },
        "unordered-list-item": Object {
          "element": "li",
          "wrapper": <ul
            className="public-DraftStyleDefault-ul"
          />,
        },
        "atomic": Object {
          "element": "figure",
        },
        "unstyled": Object {
          "aliasedElements": Array [
            "p",
          ],
          "element": "div",
        },
        "header-two": Object {
          "element": "h2",
        },
        "code-block": Object {
          "element": "pre",
          "wrapper": <pre
            className="public-DraftStyleDefault-pre"
          />,
        },
        "blockquote": Object {
          "element": "blockquote",
        },
        "header-five": Object {
          "element": "h5",
        },
        "header-three": Object {
          "element": "h3",
        },
      }
    }
    blockRendererFn={[Function]}
    blockStyleFn={[Function]}
    editorState={
      EditorState {
        "_immutable": Immutable.Record {
          "allowUndo": true,
          "currentContent": Immutable.Record {
            "entityMap": Object {
              "__add": [Function],
              "__create": [Function],
              "__get": [Function],
              "__getLastCreatedEntityKey": [Function],
              "__mergeData": [Function],
              "__replaceData": [Function],
              "add": [Function],
              "create": [Function],
              "get": [Function],
              "getLastCreatedEntityKey": [Function],
              "mergeData": [Function],
              "replaceData": [Function],
            },
            "blockMap": Immutable.OrderedMap {
              "3cmdi": Immutable.Record {
                "key": "3cmdi",
                "type": "unstyled",
                "text": "",
                "characterList": Immutable.List [],
                "depth": 0,
                "data": Immutable.Map {},
              },
            },
            "selectionBefore": Immutable.Record {
              "anchorKey": "3cmdi",
              "anchorOffset": 0,
              "focusKey": "3cmdi",
              "focusOffset": 0,
              "isBackward": false,
              "hasFocus": false,
            },
            "selectionAfter": Immutable.Record {
              "anchorKey": "3cmdi",
              "anchorOffset": 0,
              "focusKey": "3cmdi",
              "focusOffset": 0,
              "isBackward": false,
              "hasFocus": false,
            },
          },
          "decorator": null,
          "directionMap": Immutable.OrderedMap {
            "3cmdi": "LTR",
          },
          "forceSelection": false,
          "inCompositionMode": false,
          "inlineStyleOverride": null,
          "lastChangeType": null,
          "nativelyRenderedContent": null,
          "redoStack": Immutable.Stack [],
          "selection": Immutable.Record {
            "anchorKey": "3cmdi",
            "anchorOffset": 0,
            "focusKey": "3cmdi",
            "focusOffset": 0,
            "isBackward": false,
            "hasFocus": false,
          },
          "treeMap": Immutable.OrderedMap {
            "3cmdi": Immutable.List [
              Immutable.Record {
                "start": 0,
                "end": 0,
                "decoratorKey": null,
                "leaves": Immutable.List [
                  Immutable.Record {
                    "start": 0,
                    "end": 0,
                  },
                ],
              },
            ],
          },
          "undoStack": Immutable.Stack [],
        },
      }
    }
    handleKeyCommand={[Function]}
    keyBindingFn={[Function]}
    onChange={[Function]}
    readOnly={false}
    spellCheck={false}
    stripPastedStyles={false}
  />
</Fragment>
`;
